### ISSUE TITLE
Support for Device Flow

### ISSUE THREAD

**Issue by matallui** [auth0/react-native-auth0#608]
### Describe the problem you'd like to have solved

My React Native app is targeted to TV devices. Therefore I'd like to follow the Device Flow in order to authenticate the TV app.
Right now it sounds like the process for that would be to manually use Axios (or similar) in order to obtain the Access Token, and then just manually save it (and manage it) for whenever we need to query a protected API endpoint.

### Describe the ideal solution

It would be nice to have a hooks similar to what is done in Next.js (e.g., `useUser()`, `withPageAuthRequired()`) so we could easily implement this authorization mechanism and also keep track of the session.

## Alternatives and current work-arounds

It sounds to me that we'd need to manually use Axios (or `fetch`) to obtain the Access Token, then manage the session manually.

### Additional information, if any

My backend API is running as a Next.js application, so I'd like both apps (Next.js and Expo) to use the same API via tRPC and get a seamless experience when it comes to authenticated API endpoints.
So, basically I'd like to make sure I have a way to store the session/user information in the tRPC context to be used across all my React/React Native applications.

Ideally this would be a set of React hooks/functions that could be used not only in React Native, but also in React applications made specifically for TV devices (e.g., Tauri apps).

**Comment by poovamraj** [comment]
Hi @matallui, we have received request for this feature sparsely before. We will check if this is something we would like to add to the SDK or guide users with our quickstart. Will keep you in the loop about the way forward.

**Comment by poovamraj** [comment]
Hi @matallui, Thanks a lot for your feedback

I spoke about this internally and we are planning to see the usage and requirement for this feature. As of now we don't have plans implementing this in the near future. Please feel free to add your comments and we will take them into consideration. We will close this issue for now.

**Comment by bednar** [comment]
From my perspective, this would also be beneficial, as I am currently working on apps for AndroidTV.

**Comment by jmarek-sky** [comment]
Same here, it would be great to have the device flow supported by the RN SDK. Without it we are forced to manage the user tokens "manually" and reinvent the wheel with handling token refreshes etc.
Something that could alleviate the pain without fully supporting the device flow, would be to make it possible to "hand over" the management of the tokens to Auth0 SDK once we obtained them manually. This would mean we could leverage the `getCredentials()` method and its ability to transparently refresh the token behind the scenes for us.

**Comment by jmarek-sky** [comment]
I'll answer my own question. Maybe it will help someone else.
There's the [`saveCredentials`](https://auth0.github.io/react-native-auth0/classes/Types.CredentialsManager.html#saveCredentials) method which can be used to handover management of the tokens to Auth0 SDK once you obtained them in a flow outside of the SDK itself.
How did I miss that before? 🤔

**Comment by brushton-ces** [comment]
Another vote for this use case.  @poovamraj has there been any movement with out of the box support for device flow in React Native?
